pypqlx.py

```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
+            a = pdf.power[max(pdf.count[pdf.period == i])]
+            mode.append(a)
+        return
+        
     def median(self):
         pass
     
     def average(self):
-        pass
+        avg = []
+        for i in sorted(set(pdf.period)):
+            a = sum(pdf.power[pdf.period == i] * pdf.count[pdf.period == i]) / sum(pdf.count[pdf.period == i])
+            avg.append(a)
+        return
     
     def min(self):
-        pass
+        minimun = []
+        for i in sorted(set(pdf.period)):
+            a = min(pdf.power[pdf.period == i])
+            minimun.append(a)
+        return
     
     def max(self):
-        pass
+        maximun =[]
+        for i in sorted(set(pdf.period)):
+            a = max(pdf.power[pdf.period == i])
+            maximun.append(a)
+        return
```

Comment:
You must return something, or better, a numpy array of the maximun values.

Reply:
Sorry, I fixed it.